File: Problems/Mine approach/Bit Manipulation/cpp/Numbers of 1 Bits.cpp
// Write a function that takes an integer and returns the number of 1 bits it has.
int Solution::numSetBits(int A) {
    int total_ones = 0;
    while (A != 0) {
        A = A & (A-1);
        total_ones++;
    }
    return total_ones;
}

File: Problems/Mine approach/Bit Manipulation/js/Numbers of 1 Bits.js
// Write a function that takes an integer and returns the number of 1 bits it has.
module.exports = { 
    //param A : integer
    //return an integer
       numSetBits : function(A){
    
       let B, count =0;
       B = A.toString(2);
       for(let i=0; i<B.length; i++){
           if(B.charAt(i) === "1"){
               count++;
           }
       }
       // console.log(B.charAt(B.length-1));
       // console.log(B);
       return count;
   }
       
   };